assembl/graphql/vote_session.py

```diff
@@ -56,16 +57,19 @@ def mutate(root, args, context, info):
         if discussion_phase is None:
             raise Exception(
                 "A vote session requires a discussion phase, check discussionPhaseId value")
-        phase_slug = "voteSession"
-        if discussion_phase.identifier != phase_slug:
+        phase_identifier = "voteSession"
+        if discussion_phase.identifier != phase_identifier:
             raise Exception(
-                "A vote session can only be created or edited with a '{}' discussion phase, check discussionPhaseId value".format(phase_slug))
+                "A vote session can only be created or edited with a '{}' discussion phase, check discussionPhaseId value".format(phase_identifier))
 
-        # TODO: see if we can avoid this next(iter( thing with a one-to-one relationship
-        vote_session = next(iter(discussion_phase.vote_session or []), None)
+        vote_session = discussion_phase.vote_session
         if vote_session is None:
-            vote_session = models.VoteSession(
-                discussion_phase=discussion_phase)
+            require_cls_permission = make_cls_permissions_querier(models.VoteSession, context)
```

Comment:
maybe name it `check_vote_session_permission` ?

Reply:
Agreed, could be renamed to this instead, and maybe not even be a closure.